TRANSFORMERS/MATRIX_MANIPULATION/DOT_PRODUCT/DOT_PRODUCT.py

```diff
@@ -0,0 +1,25 @@
+import numpy as np
+from flojoy import flojoy, Matrix
+
+
+@flojoy
+def DOT_PRODUCT(a: Matrix, b: Matrix) -> Matrix:
+    """The DOT_PRODUCT node takes two input matrices, multiplies them
```

Comment:
What kinda of error will be throw if the 2 matrix has incompatible dimension? Can we do a check and throw a custom error message so that will be more readable from the one from numpy I assume.